wdl-engine/src/backend/local.rs

```diff
@@ -269,39 +271,78 @@ impl TaskExecutionBackend for LocalTaskExecutionBackend {
         None
     }
 
-    fn localize_inputs<'a, 'b, 'c, 'd>(
+    fn localize_inputs<'a, 'b, 'c>(
         &'a self,
-        downloader: &'b dyn Downloader,
+        downloader: Arc<dyn Downloader>,
         inputs: &'c mut [Input],
-    ) -> BoxFuture<'d, Result<()>>
+    ) -> BoxFuture<'c, Result<()>>
     where
-        'a: 'd,
-        'b: 'd,
-        'c: 'd,
-        Self: 'd,
+        'a: 'c,
+        'b: 'c,
+        Self: 'c,
     {
-        async {
-            for input in inputs {
-                // TODO: parallelize the downloads
-                let location = match input.path() {
-                    EvaluationPath::Local(path) => Location::Path(path.into()),
-                    EvaluationPath::Remote(url) => downloader
-                        .download(url)
-                        .await
-                        .map_err(|e| anyhow!("failed to localize `{url}`: {e:?}"))?,
-                };
-
-                let guest_path = location
-                    .to_str()
-                    .with_context(|| {
-                        format!("path `{path}` is not UTF-8", path = location.display())
-                    })?
-                    .to_string();
-
-                // Set the guest path to the download location for path translation
-                let location = location.into_owned();
-                input.set_guest_path(guest_path);
-                input.set_location(location);
+        async move {
```

Comment:
We should obviously come back and get this one too once we settle on a solution in `crankshaft.rs`.